app/controllers/roles_controller.rb

```diff
@@ -66,6 +66,21 @@ def destroy
     redirect_to controller: 'plans', action: 'share', id: @role.plan.id
   end
 
+  def archive
```

Comment:
I'm not 100% on the `archive` action name for this. Perhaps `deactivate` is a bit clearer on what it does - "remove" is certainly not accurate (even though it eventually refers to "remove plan" action)